.github/push-image.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+set -e
+
+if [ "$#" -ne 2 ]; then
+    echo "Pushes a libero container image to Docker Hub with tags"
+    echo 
+    echo "Usage: $0 project input_tag"
+    echo "Example: $0 dummy-api 6cd9d7ebad4f16ef7273a7a831d79d5d5caf4164"
+    echo "Relies on the following environment variables:"
+    echo "- GITHUB_REF, GITHUB_SHA (GH Action default)"
+    echo "- DOCKER_USERNAME, DOCKER_PASSWORD"
+    exit 1
+fi
+
+project="$1"
+image_tag="$2"
+
+
+# login
+DOCKER_REGISTRY="${DOCKER_REGISTRY:-}"
+echo "${DOCKER_PASSWORD}" | docker login --username "${DOCKER_USERNAME}" --password-stdin "${DOCKER_REGISTRY}"
+
+# tag temporarily as liberoadmin due to lack of `libero/` availability
+name="${DOCKER_REGISTRY}liberoadmin/${project}"
+input_image="libero/${project}:${image_tag}"
+
+if [[ "$GITHUB_REF" == "refs/heads"* ]]; then
+    # push `branch-sha` tagged image
+    branch="${GITHUB_REF/refs\/heads\//}"
+    docker tag "$input_image" "${name}:${branch}-${GITHUB_SHA}"
+    docker push "${name}:${branch}-${GITHUB_SHA}"
+
+    if [[ "$branch" = "master" ]]; then
+        # push `latest` tag
+        docker tag "$input_image" "${name}:latest"
+        docker push "${name}:latest"
+    fi
+
+elif [[ "$GITHUB_REF" =~ refs/tags/v(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*) ]]; then
```

Comment:
a similar scenario to check is this sequence:
- release `1.0.0`
- release `1.1.0`
- release `1.0.1` with a bugfix
Now `1` points to `1.0.1` rather than `1.1.0`. I think semantic versioning makes this case invalid as you should always bugfix on the latest minor due to the backward compatibility, making this a non-issue. Does all of this make sense?

Reply:
So I had a similar problem in my last role, and I spent some time writing a shell script to figure out if one version number was greater than the other, and to only move the link if it was. It was never needed, as in we never had a situation where we had to go back and build a 1.2.3 after building a 1.3.1 or similar. It's a real problem that 'could' happen, but I just don't see it being an issue for us at the moment and hence I'd note (maybe in a new issue) that the work needs doing and then defer it until a time when we need it.

Reply:
I have added this topic to the sync agenda. I would suggest not addressing it in this PR or the libero/script but in the process and practices of the teams.